.github/workflows/ci.yml

```diff
@@ -46,9 +46,27 @@ jobs:
       - name: Install Dependencies
         run: .ci/scripts/install-deps.sh
 
-      - name: Test
+      - name: Set up publishing to maven central
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+
+      - name: Install gpg key
+        run: |
+          cat <(echo -e "${{ secrets.OSSRH_GPG_SECRET_KEY }}") | base64 --decode | gpg --batch --import
+
+      - name: Test and Publish
```

Comment:
Add a check that `-SNAPSHOT` is mandatory for version, otherwise if someone removes `-SNAPSHOT` from `pom.xml` it will get released to maven central